discovery_engine_core/providers/src/client.rs

```diff
@@ -45,12 +45,11 @@ pub enum Error {
     ),
 }
 
-/// Client that can provide documents.
-#[derive(Default)]
-pub struct Client {
-    token: String,
-    url: String,
-    timeout: Duration,
+/// Represents a Query to Newscatcher.
+///
+/// Should not be implemented outside of this module.
```

Comment:
We can seal it.

Reply:
`Query` is defacto sealed because it's in a private module and not reexported. having an additional `Query: Seal` sealing trait bound makes it more explicit, which is better i guess.

if we seal `Query` explicitely, we should also reexport it in the root module to allow for better docs generation, because it is used in a public interface.